### PR TITLE
1708 BE Tyjoo Scrabble

### PR DESCRIPTION
**Pull request by tjp2021**

### CODE REVIEW

File: lib/scrabble.rb [hunk 1]
```diff
@@ -15,4 +13,81 @@ def point_values
       "Y"=>4, "Z"=>10
     }
   end
+
+
+  def score(word)
+    if word == nil
+      return 0
+    elsif word.is_a? Integer
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:21]: I like that you're looking out for edge cases! This wouldn't quite catch all cases though (what if I passed in a symbol or a hash, etc.?). Think about ways you could check that it's not a string or that it only includes alphabet letters.

File: lib/scrabble.rb [hunk 2]
```diff
@@ -15,4 +13,81 @@ def point_values
...
+      return 0
+    else
+      data = word.upcase.split("")
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:24]: Looks like you use the 'something.upcase.split("")' multiple times in your code. Consider pulling that logic into its own method to make your code more dry.

File: lib/scrabble.rb [hunk 3]
```diff
@@ -15,4 +13,81 @@ def point_values
...
+      score_conversion_string(data)
+    end
+  end
+
+  def score_conversion_string(data)
+    if data == []
+      return 0
+    else
+      data.map do |letter|
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:33]: I like that you're using `#map` & not `#each`! This block is also used in the method below. Try to refactor this out to make increase dry & clean up method length.

File: lib/scrabble.rb [hunk 4]
```diff
@@ -15,4 +13,81 @@ def point_values
...
+
+
+  def convert_letters_to_array(data)
+    data.map do |letter|
+      @key[letter]
+    end
+  end
+
+
+  def score_with_multipliers(data, multipliers, double = 1)
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:47]: This method is pretty lengthy. Looks like you were working on it up until the end of time, but I'd like to see you refactor out those nested blocks.

File: lib/scrabble.rb [hunk 5]
```diff
@@ -15,4 +13,81 @@ def point_values
...
+    data = data.upcase.split("")
+    if data.count >= 7
+      new_array = convert_letters_to_array(data)
+      new_array.zip(multipliers).map do |x,y|
+        x * y
+      end.inject(:+) + 10 * double
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:53]: If you're using Ruby 2.4, you can use `#sum` instead of `#inject`; it makes the code a bit more readable

File: lib/scrabble.rb [hunk 6]
```diff
@@ -15,4 +13,81 @@ def point_values
...
+      end.inject(:+) * double
+    end
+  end
+
+
+
+
+
 end
+
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:67]: Make sure to remove comments from pushed code!

File: test/scrabble_test.rb [hunk 7]
```diff
@@ -6,8 +6,58 @@
...
+    scrabble = Scrabble.new
+    assert_equal 8, scrabble.score("hello")
+  end
+
+  def test_it_can_score_a_different_string
+    scrabble = Scrabble.new
+    assert_equal 13, scrabble.score("sparkle")
+  end
+
+  def test_empty_string_is_zero
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:23]: Nice breaking out multiple tests for each edge case!

File: test/word_list_test.rb [hunk 8]
```diff
@@ -6,4 +6,14 @@
 
 class WordListTest < Minitest::Test
 
+
+  def test_map_letters_to_one
```
Comment (VictoriaVasys) [inline test/word_list_test.rb:10]: Were you attempting to do something here? Not sure where this is going.

**Review by VictoriaVasys — COMMENTED** [review]
Nice job overall TJ! Make sure you're working to extract out logic (one method was particularly long). Also, be careful not to push comments or extraneous files to your final code. We'd also like to see you committing every 15-20 minutes for an exercise like this, so try to work towards that on future timed exercises. 

Way to almost get full functionality, break out some logic & think about edge cases!
